L1Trigger/L1THGCal/src/HGCalTriggerCellCalibration.cc

```diff
@@ -44,25 +60,52 @@ void HGCalTriggerCellCalibration::calibrateInMipT(l1t::HGCalTriggerCell& trgCell
 
 void HGCalTriggerCellCalibration::calibrateMipTinGeV(l1t::HGCalTriggerCell& trgCell) const {
   const double MevToGeV(0.001);
+  double trgCellEt(0.);
 
-  DetId trgdetid(trgCell.detId());
-  unsigned trgCellLayer = triggerTools_.layerWithOffset(trgdetid);
+  unsigned trgCellLayer = triggerTools_.layerWithOffset(trgCell.detId());
 
-  if (dEdX_weights_.at(trgCellLayer) == 0.) {
-    throw cms::Exception("BadConfiguration")
-        << "Trigger cell energy forced to 0 by calibration coefficients.\n"
-        << "The configuration should be changed. "
-        << "Discarded layers should be defined in hgcalTriggerGeometryESProducer.TriggerGeometry.DisconnectedLayers "
-           "and not with calibration coefficients = 0\n";
-  }
+  if (DetId(trgCell.detId()).det() == DetId::HGCalTrigger and
+      HGCalTriggerDetId(trgCell.detId()).subdet() == HGCalTriggerSubdetector::HFNoseTrigger) {
```

Comment:
Definitely it should be put in the trigger tools

Reply:
yes, done